skimage/transform/_geometric.py

```diff
@@ -921,7 +921,12 @@ def __init__(self, matrix=None, scale=None, rotation=None, shear=None,
 
     @property
     def scale(self):
-        return np.sqrt(np.sum(self.params ** 2, axis=0))[:self.dimensionality]
+        if self.dimensionality != 2:
+            return np.sqrt(np.sum(self.params ** 2, axis=0))[:self.dimensionality]
+        else:
+            ss = np.sum(self.params ** 2, axis=0)[:self.dimensionality]
+            ss[1] = ss[1] / (math.tan(self.shear)**2 + 1)
```

Comment:
Could you help me understand what is happening here? Why do we suddenly need a different calculation for the 2D case? If the matrix in `self.params` is correct, the dimensionality shouldn't matter?

Also you have moved the the indexing with `[:self.dimensionality]` into the square root's argument; was that intentional?

Reply:
We are not allowed to access `self.shear` in other dimensionality since the shear parameter is supported only in the case of the 2D version.
See:
https://github.com/scikit-image/scikit-image/blob/de46b0fb265fadb75e796836c73fb00923cad7a3/skimage/transform/_geometric.py#L934-L941

> Also you have moved the the indexing with [:self.dimensionality] into the square root's argument; was that intentional?

The result is identical. But I fixed this not to be confusing.

Reply:
Ah, I didn't notice that `self.shear` was only accessed in one case. However, I'm still confused. Why doesn't the 
```python
np.sqrt(np.sum(self.params ** 2, axis=0))[:self.dimensionality]
```
work for the 2D case? Testing with 
```python
t = ski.transform.AffineTransform(translation=(3, 4), rotation=1, scale=(5, 6), shear=0.2)
np.sqrt(np.sum(t.params ** 2, axis=0))[:t.dimensionality]
# array([5.        , 6.12203307])
```
seems to return a wrong factor, although it's suspiciously not far off. `t.scale` (your calculation in 2a5e6f9ac2ede374e3a789ae027eb4d69eb9f304) returns the correct result. I'm asking myself if the calculation is even correct for the other non-2D cases... :thinking:

Reply:
OK. With the new definition of the `self.params`, I used the following equations:
```
# (pseudo code)

# For sx
params[0, 0] := sx * cos(rot)
params[1, 0] := sx * sin(rot)

# Calc square sum
sum(params[:, 0]**2) 
-> sx**2 (cos(rot)**2 + sin(rot)**2)
-> sx**2

# Solving by sx
sx = sqrt(sum(params[:, 0]**2))

# For sy
params[0, 1] := -sy * (tan(shear) * cos(rot) + sin(rot))
params[1, 1] := -sy * (tan(shear) * sin(rot) - cos(rot))

# the square sum:
sum(params[:, 1]**2)
-> sy**2 (cos(rot)**2 + sin(rot)**2 + tan(shear)**2(cos(rot)**2 + sin(rot)**2))
-> sy**2 (1 + tan(shear)**2)

# Solving by sy
sy = sqrt(sum(params[:, 1]**2) / (1 + tan(shear**2)))
```

Reply:
That deduction seems correct. :+1: 

However, that the calculation for the 2D case is different from the "general ND" case leads me to believe that the latter may be wrong? Additionally, this property doesn't seem to be tested at all for non-2D cases.

@jni, you seem to have added ND support in #5188. Do you have more context concerning this calculation for ND matrices? https://github.com/scikit-image/scikit-image/blob/f4978b11499e499cc006c417b3bf0ccf245ca72c/skimage/transform/_geometric.py#L924

I don't consider this blocking for this PR. For the 2D case at least, I'm reasonably sure that this PR is "more" correct. For other cases it leaves the old behavior unchanged.